File: WitchcraftApp/src/main/java/com/tarbadev/witchcraft/Item.java
package com.tarbadev.witchcraft;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Builder
@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity(name = "items")
public class Item {
    @Id
    @GeneratedValue(strategy= GenerationType.AUTO)
    private Integer id;
    private String name;
    private Double quantity;
    private String unit;

  public Item addQuantity(Double quantity) {
    this.quantity += quantity;
    return this;
  }
}

File: WitchcraftApp/src/main/kotlin/com/tarbadev/witchcraft/carts/domain/entity/Item.kt
package com.tarbadev.witchcraft.carts.domain.entity

import com.tarbadev.witchcraft.converter.getUnitShortName
import com.tarbadev.witchcraft.recipes.domain.entity.Ingredient

data class Item(
    val id: Int = 0,
    val name: String,
    val quantity: Double,
    val unit: String,
    val enabled: Boolean
) {
  companion object {
    fun fromIngredient(ingredient: Ingredient): Item {
      return Item(
          0,
          ingredient.name,
          ingredient.quantity.getValue().toDouble(),
          getUnitShortName(ingredient.quantity),
          true
      )
    }
  }
}
